ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/managed-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-managed-identity-provision-chain.yaml

```diff
@@ -14,6 +12,8 @@ chain:
   - ref: ipi-install-rbac
   - ref: openshift-cluster-bot-rbac
   - ref: ipi-install-hosted-loki
+  - ref: azure-provision-service-principal-minimal-permission
+  - ref: azure-provision-bastionhost-managed-identity
```

Comment:
Seem like `azure-provision-bastionhost-managed-identity` does not depend on `azure-provision-service-principal-minimal-permission`, while  `ipi-conf-azure-oidc-creds-provision` depends on `azure-provision-service-principal-minimal-permission` when `ENABLE_MIN_PERMISSION_FOR_STS` enabled, so I think `azure-provision-service-principal-minimal-permission` need to be prior to `ipi-conf-azure-oidc-creds-provision`

Reply:
The issue also happened on other oidc + mini-perm install workflow.

Reply:
good catch, I moved step `azure-provision-service-principal-minimal-permission` before step `ipi-conf-azure-oidc-creds-provision`.

> Seem like azure-provision-bastionhost-managed-identity does not depend on azure-provision-service-principal-minimal-permission

yes, there is no dependency now, since there is some issues with managed identity assigned minimal permissions, need to debug further and update step later to support it.